pdfgen/src/types/hierarchy/primitives/font.rs

```diff
@@ -30,23 +33,40 @@ impl Font {
         FONT,
         SUBTYPE,
         BASE_FONT,
+        NAME,
     }
 
-    /// Create a new Font object with the provided id, subtype and base_font.
-    pub fn new<S, B>(id: ObjId, subtype: S, base_font: B) -> Self
+    /// Create a new [`Font`] object with the provided id, subtype and base_font.
+    pub fn new<N, S, B>(name: N, id: ObjId, subtype: S, base_font: B) -> Self
     where
+        N: Into<Vec<u8>>,
         S: Into<Vec<u8>>,
         B: Into<Vec<u8>>,
     {
+        let name = Name::new(name.into());
         let subtype = Name::new(subtype.into());
         let base_font = Name::new(base_font.into());
 
         Font {
             id,
+            name,
             subtype,
             base_font,
         }
     }
+
+    /// Writes the PDF Font reference, using the FontName and provided size.
+    pub fn write_ref(
+        &self,
+        size: u32,
+        writer: &mut dyn std::io::Write,
+    ) -> Result<usize, std::io::Error> {
+        // /FName size
+        Ok(pdfgen_macros::write_chain! {
+            self.name.write(writer),
+            writer.write(format!{" {}", size}.as_bytes()),
```

Comment:
Reminder for our future selves: 

We should replace these `write(format!(...))` calls with `write!(...)` calls to avoid unnecessary `String` allocation when using `format!` macro.

Reply:
Opened an issue for this https://github.com/pediferrous/pediferrous/issues/57. Will resolve the comment for now.